libsolidity/analysis/PostTypeChecker.cpp

```diff
@@ -336,27 +342,38 @@ struct EventOutsideEmitErrorOutsideRevertChecker: public PostTypeChecker::Checke
 		if (*_functionCall.annotation().kind == FunctionCallKind::FunctionCall)
 			if (auto const* functionType = dynamic_cast<FunctionType const*>(_functionCall.expression().annotation().type))
 			{
+				if (!m_inRequire && functionType->kind() == FunctionType::Kind::Require)
+					m_inRequire = true;
```

Comment:
We might also instead reuse ``m_currentStatement`` here, i.e. set that here to the outer call? Well, not sure which is better.
I haven't fully thought this through in nested cases yet - might be worthwhile to check for pathological things like ``revert SomeError(require(false,AnotherError()));`` or so - none of this should be valid, but worthwhile to think about whether there can be a case that slips through.

Reply:
Sure; the reason I did it this way is precisely because I could think of a situation where a require would be within a nested call.